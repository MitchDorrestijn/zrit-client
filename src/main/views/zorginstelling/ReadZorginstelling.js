/**
 * React related imports
 */
import React from 'react';

/**
 * Self made components imports
 */
import TextContent from '../../components/TextContent';
import ZorginstellingTable from '../../components/ZorginstellingTable';
import {checkIfUserIsAdmin} from '../../global/Methods';

/**
 * This component puts the textblock and zorginstellingtable together and renders it
 */
const ReadZorginstelling = (props) => {
  return (
		<div>
	    <TextContent title={checkIfUserIsAdmin() ? "Beheer zorginstellingen" : "Uw zorginstelling"}>
        {checkIfUserIsAdmin() ?
          <div>
            Hieronder kunt u zorginstellingen beheren.
            Druk op het rondje vooraan de tabel om een record te selecteren,
            vervolgens kunt u het record bewerken door op de knop 'bewerken' te drukken.
            Om een nieuw item toe te toevoegen drukt u op de knop 'Toevoegen'.
          </div>
          :
          <div>
            Hieronder ziet u uw zorginstelling(en). Als u deze wil bewerken of verwijderen moet u de superbeheerder inschakelen.
          </div>
        }
	    </TextContent>
	    <ZorginstellingTable routes={props.routes} history={props.history}/>
  	</div>
	);
}

export default ReadZorginstelling;
