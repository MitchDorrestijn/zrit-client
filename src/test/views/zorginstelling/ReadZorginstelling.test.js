import React from 'react';
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';
import ReadZorginstelling from '../../../main/views/zorginstelling/ReadZorginstelling';

const wrapper = shallow(<ReadZorginstelling />);

test('<ReadZorginstelling /> should match snapshot', () => {
  expect(toJSON(wrapper)).toMatchSnapshot();
});

test('<ReadZorginstelling /> should render one <TextContent> component', () => {
  expect(wrapper.find('TextContent').length).toBe(1);
});

test('<ReadZorginstelling /> should render one <ZorginstellingTable> component', () => {
  expect(wrapper.find('ZorginstellingTable').length).toBe(1);
});
