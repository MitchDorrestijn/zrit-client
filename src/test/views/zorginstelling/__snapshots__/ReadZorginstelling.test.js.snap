// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReadZorginstelling /> should match snapshot 1`] = `
<div>
  <TextContent
    title="Zorginstellingen beheren"
  >
    Hieronder kunt u zorginstellingen beheren. Druk op het rondje vooraan de tabel om een record te selecteren, vervolgens kunt u het record bewerken door op de knop 'bewerken' te drukken. Om een nieuw item toe te toevoegen drukt u op de knop 'Toevoegen'.
  </TextContent>
  <ZorginstellingTable />
</div>
`;
